Code/Framework/AzToolsFramework/AzToolsFramework/API/EditorViewportIconDisplayInterface.h

```diff
@@ -60,9 +60,23 @@ namespace AzToolsFramework
             Error
         };
 
+        //! @deprecated use AddIcon and then DrawIcons to batch correctly.
```

Comment:
Maybe use the AZ_DEPRECATED macro to get a compilation warning

Reply:
A compilation warning results in a compilation error since we use warnings-as-errors..  Not sure if there's a way to warn without stopping compilation

Reply:
https://github.com/o3de/community/blob/main/guides/o3de-deprecation-guidelines.md

I don't really actually want to change the API here.  So I'm not sure what to do.  AZ_DEPRECATED would immediately cause all compilation to *fail* which uses it, so I don't want that.  And I don't actually need to change the API so don't really have plans to actually deprecate this function. 

What I'll do is NOT deprecate it, but add a comment to the api to consider using the other ones.

Reply:
Yeah sounds like the right way to go !